### PR TITLE
Remove unused mono_pages_not_faulted, autoconf for mlock instead of mincore.

### PR DESCRIPTION
**Pull request by jaykrell**

### CODE REVIEW

File: support/sys-mman.c [hunk 1]
```diff
@@ -89,7 +89,7 @@ Mono_Posix_Syscall_msync (void *start, mph_size_t len, int flags)
 int
 Mono_Posix_Syscall_mlock (void *start, mph_size_t len)
 {
-#if !defined(HAVE_MINCORE)
+#if !defined(HAVE_MLOCK)
```
Comment (jaykrell) [inline support/sys-mman.c]: Or __HAIKU__ like munlock?

Comment (NattyNarwhal) [reply support/sys-mman.c]: i'd recommend just testing for mlock/munlock, and that should cover Haiku

**Comment by jaykrell** [comment]
Just one test for both, eh? Splitting them up would kinda be nonsense?
I’ll do that.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by lewurm** [comment]
@monojenkins squash

**Pull request merged**